compiler/rustc_error_codes/src/error_codes/E0793.md

```diff
@@ -0,0 +1,49 @@
+A lifetime parameter of a function definition is called *late-bound* if it both:
+
+1. appears in an argument type
+2. does not appear in a where clause
+
+Function types are not generic over late-bound lifetime parameters,
+but are universally quantified over them. This means that when referring to a
+function of a concrete type by substituting concrete types for
+generic ones, you cannot specify a late-bound lifetime, because such lifetime
+parameters are resolved at the function's call site(s).
```

Comment:
```suggestion
concrete instance of the generic function, you cannot specify any of
its late-bound lifetime parameters, because they are not resolved until
each call site(s).
```